core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -350,6 +354,224 @@ public void testPartitionsTableScanNotNullFilter() {
     validateIncludesPartitionScan(tasksUnary, 3);
   }
 
+  @Test
+  public void testFilesTableScanNoFilter() {
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_0)
+        .commit();
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_1)
+        .commit();
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_2)
+        .commit();
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_3)
+        .commit();
+
+    Table dataFilesTable = new DataFilesTable(table.ops(), table);
+    Types.StructType expected = new Schema(
+        required(102, "partition", Types.StructType.of(
+            optional(1000, "data_bucket", Types.IntegerType.get())),
+            "Partition data tuple, schema based on the partition spec")).asStruct();
+
+    TableScan scanNoFilter = dataFilesTable.newScan().select("partition.data_bucket");
+    Assert.assertEquals(expected, scanNoFilter.schema().asStruct());
```

Comment:
I'm not sure you need to add the schema check since this is trying to validate pushdown, but I'm fine keeping it.